pynicotine/shares.py

```diff
@@ -367,75 +368,73 @@ def is_hidden(folder, filename=None, entry=None):
 
         return False
 
-    def get_files_list(self, folder, oldmtimes, oldfiles, oldstreams, rebuild=False, folder_stat=None):
+    def get_files_list(self, shared_folder, oldmtimes, oldfiles, oldstreams, rebuild=False):
         """ Get a list of files with their filelength, bitrate and track length in seconds """
 
-        if folder_stat is None:
-            folder_stat = os.stat(encode_path(folder))
-
-        folder_unchanged = False
-        virtual_folder = self.real2virtual(folder)
-        mtime = folder_stat.st_mtime
-
-        file_list = []
         files = {}
         streams = {}
-        mtimes = {folder: mtime}
+        mtimes = {}
+        folders = deque([(shared_folder, None)])
 
-        if not rebuild and folder in oldmtimes and mtime == oldmtimes[folder]:
-            try:
-                files[virtual_folder] = oldfiles[virtual_folder]
-                streams[virtual_folder] = oldstreams[virtual_folder]
-                folder_unchanged = True
+        while folders:
+            folder, folder_stat = folders.pop()
```

Comment:
Is it intentional that the while loop exits before the last item in the deque?

Reply:
Are you sure? That shouldn't be the case.

Reply:
It only from reviewing the code, that it appears like the code after this statement will be unreachable for the final item.

However, no share items appear to be missed. I must be misunderstanding.

Reply:
while loops don't terminate immediately when the condition becomes false, they terminate when `while folders` is evaluated again.